File: DiKErnel/Src/DiKErnel.DomainLibrary/Public/Defaults/GrassRevetmentWaveImpact/GrassRevetmentWaveImpactOpenSodTopLayerDefaults.cpp
// Copyright (C) Stichting Deltares 2022. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include "GrassRevetmentWaveImpactOpenSodTopLayerDefaults.h"

namespace DiKErnel::DomainLibrary
{
    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetTimeLineAgwi() const
    {
        return 0.8;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetTimeLineBgwi() const
    {
        return -0.00001944;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetTimeLineCgwi() const
    {
        return 0.25;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetMinimumWaveHeightTemax() const
    {
        return 3600000;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetMaximumWaveHeightTemin() const
    {
        return 3.6;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetWaveAngleImpactNwa() const
    {
        return 2.0 / 3.0;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetWaveAngleImpactQwa() const
    {
        return 0.35;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetWaveAngleImpactRwa() const
    {
        return 10;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetUpperLimitLoadingAul() const
    {
        return 0;
    }

    double GrassRevetmentWaveImpactOpenSodTopLayerDefaults::GetLowerLimitLoadingAll() const
    {
        return 0.5;
    }
}

File: Src/DiKErnel.DomainLibrary/Defaults/GrassRevetmentWaveImpact/GrassRevetmentWaveImpactOpenSodTopLayerDefaults.cs
﻿// Copyright (C) Stichting Deltares and State of the Netherlands 2023. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify it under the terms of the
// GNU Lesser General Public License as published by the Free Software Foundation, either
// version 3 of the License, or (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program. If not, see <http://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

namespace DiKErnel.DomainLibrary.Defaults.GrassRevetmentWaveImpact
{
    /// <summary>
    /// Class that contains default values for grass revetment wave impact with an open sod
    /// top layer.
    /// </summary>
    public class GrassRevetmentWaveImpactOpenSodTopLayerDefaults : IGrassRevetmentWaveImpactTopLayerDefaults
    {
        public double TimeLineAgwi => 0.8;
        
        public double TimeLineBgwi => -0.00001944;
        
        public double TimeLineCgwi => 0.25;
    }
}